File: src/main/scala/com/barley/respository/AttachmentRepository.scala
package com.barley.respository

import java.lang.{Iterable, Long}

import com.barley.model.Attachment
import org.springframework.data.repository.CrudRepository
import org.springframework.stereotype.Component

/**
  * Created by vikram.gulia on 11/25/15.
  */
@Component
trait AttachmentRepository extends CrudRepository[Attachment, Long] {
  def findByMessageId(messageId: Long): Iterable[Attachment]
}

File: src/main/java/com/barley/repository/AttachmentRepository.java
package com.barley.repository;

import com.barley.model.Attachment;
import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Component;

import java.util.List;

@Component
public interface AttachmentRepository extends CrudRepository<Attachment, Long> {
    List<Attachment> findByMessageId(Long messageId);
}